frontend/src/views/Lobby.tsx

```diff
@@ -65,36 +66,38 @@ function LobbyPage() {
 
   // Grab the nickname variable and add the user to the lobby.
   useEffect(() => {
-    // Grab the nickname variable; otherwise, set an error.
-    if (location && location.state && location.state.nickname) {
-      setNickname(location.state.nickname);
+    // Grab the user and room information; otherwise, redirect to the join page
+    if (location && location.state && location.state.user && location.state.room) {
+      setUser(location.state.user);
+      setRoom(location.state.room);
```

Comment:
What if the `user` is present but the `room` isn't? Or vice versa?

If you throw an error instead of redirecting, then you could throw an error depending on whether the `user` or `room` isn't provided; otherwise, no change is necessary.

Reply:
Not sure, my thought was just that if any of those are missing, then the user hasn't joined the room yet and needs to re-join to be allowed back on the lobby page. However I can def look into that after discussisng/making the other changes you suggested!

Reply:
Since it's the same action of redirect right now, I think this is fine. Later, I think it may make sense to redirect with a message on why they weren't connected to the lobby. If you wanted to do this in this PR, that works, but it's also fine to leave it as-is and get to that later.